contracts/transfer/circuits/src/dusk_contract/execute.rs

```diff
@@ -0,0 +1,1378 @@
+            size: 0,
+            pi_constructor: None,
+        }
+    }
+
+    fn add_circuit_public_inputs(circuit: &ExecuteCircuit, crossover_commitment: AffinePoint, fee: BlsScalar, pi: &mut Vec<PublicInput>) {
+            circuit
+            .input_poseidon_branches
+            .as_ref()
+            .unwrap()
+            .iter()
+            .for_each(|branch| {
+                pi.push(PublicInput::BlsScalar(-branch.root, 0));
+            });
+            circuit
+            .nullifiers
+            .as_ref()
+            .unwrap()
+            .iter()
+            .for_each(|nullifier| {
+                pi.push(PublicInput::BlsScalar(-nullifier, 0));
+            });
+
+            pi.push(PublicInput::AffinePoint(crossover_commitment, 0, 0));
+
+            circuit
+            .obfuscated_commitment_points
+            .as_ref()
+            .unwrap()
+            .iter()
+            .for_each(|commitment_point| {
+                pi.push(PublicInput::AffinePoint(*commitment_point, 0, 0));
+            });
+
+            pi.push(PublicInput::BlsScalar(fee, 0));
+
+    }
+
+    #[test]
+    // This test ensures the execute gadget is done correctly
+    // by creating two notes and setting their field values
+    // in the execute circuit
+    fn test_execute() -> Result<()> {
+        // Generate the (a,b) for the note
+        let secret1 = JubJubScalar::from(100 as u64);
+        let secret2 = JubJubScalar::from(200 as u64);
+        // Declare the secret spend key for the note
+        let ssk1 = SecretSpendKey::new(secret1, secret2);
+        // Derive the public spend key from the note from the above secret spend key
+        let psk1 = PublicSpendKey::from(ssk1);
+        // Assign the value of the note
+        let value1 = 600u64;
+        let r1 = JubJubScalar::from(150 as u64);
+        let nonce1 = JubJubScalar::from(350 as u64);
+        let input_note_blinder_one = JubJubScalar::from(100 as u64);
+        // Create a deterministic note so that we can assign the blinder and not have inner randomness
+        let mut note1 = Note::deterministic(
+            NoteType::Transparent,
+            &r1,
+            nonce1,
+            &psk1,
+            value1,
+            input_note_blinder_one,
+        );
+        // Set the position of the note
+        note1.set_pos(0);
+        // Define the hash of the note, which is required for the preimage check
+        let note_hash1 = note1.hash();
+        let pos_a = note1.pos();
+        // Derive the one time secret key, sk_r, for the note
+        let sk_r1 = ssk1.sk_r(note1.stealth_address());
+        // Derive the one time public key, pk_r, for the note
+        let pk_r1 = note1.stealth_address().pk_r();
+        let nullifier1 = note1.gen_nullifier(&ssk1);
+        let input_note_value_one = JubJubScalar::from(value1);
+        let input_commitment_one = compute_value_commitment(
+            input_note_value_one,
+            input_note_blinder_one,
+        );
```

Comment:
Please abstract the note creation away into a function like previously mentioned - make sure as well, that you delete the variables here which are now unused, due to direct referencing of the note.